source/blockchain_details.rst

```diff
@@ -0,0 +1,431 @@
+    |  State 0  |   +-->  State 1  |   +-->  State 2  |
+    |           |   |  |           |   |  |           |
+    +-----+-----+   |  +-----+-----+   |  +-----------+
+          |         |        |         |
+          |         |        |         |
+    +-----v-----+   |  +-----v-----+   |
+    |           |   |  |           |   |
+    |  Block 0  +---+  |  Block 1  +---+
+    |           |      |           |
+    +-----------+      +-----------+
+
+
+Leaving it with a final state of:
+
+.. code-block:: json
+
+    {
+        "balance": {
+            "Alice": 0,
+            "Bob": 100
+        }
+    }
+
+From here onwards other transactions can happen, further mutating global state
+and adding new blocks to the chain. The process will run indefinitely as
+long as there is a quorum of nodes in the network to agree on blocks and
+new valid transactions are coming in.
+
+
+.. _dlt-consensus:
+
+DLT consensus
+-------------
+
+There are many different ways to achieve consensus in DLT. Two of the
+common mechanisms that are talked about in blockchain industry are Proof
+of Work and Proof of Stake.
+
+
+Proof of Work
+~~~~~~~~~~~~~
+
+Proof of work is the most often mentioned mechanism for achieving
+consensus. Proof of Work requires that a contributor do a deterministically
+difficult amount of work that is then easy to check. Bitcoin for example
+does this by making miners hash until they get the longest
+string of zeroes. This artificially slows down block creation and makes it
+computationally and thus financially expensive in the Bitcoin network.
+Anyone can mine blocks but given the current normalize difficulty it
+takes a ridiculously long time for non-specialized hardware to mine a valid
+block. This process is the only way invented by now to reliably implement
+a public permissionless consensus where anybody can participate.
+
+
+Proof of Stake
+~~~~~~~~~~~~~~
+
+Proof of stake is far more like a traditional weighted election model.
+Everyone locks up some value as a promise of their good intentions inside
+the system and then there are fixed voting rounds where each person votes
+using the weight of the value locked up. In an example both Alice and Bob
+stake 50 value into the system, they both have equal votes but neither have
+majority. Both together can vote and provide majority for confirming a block.
+Anyone can propose a block but only those with stake can vote.
+
+This model does work very well in case the permissioned DLT, i.e. where
+participants of consensus have to be given an explicit prior permission to 
+join. In this case their stake contribution can be verified as part of the 
+permission granting process.
+ 
+
+.. _dlt-consensus-details:
+
+Credits consensus
+-----------------
+
+Credits consensus is a leaderless two phase commit algorithm with variable
+:ref:`voting power <dlt-consensus-voting-power>`.
+
+This means that each and every DLT network participant is equal in it's
```

Comment:
I've given definitions to both in the blockchain_terms.rst, I'll go over and make sure all usages are compliant to that definition.